packages/plugins/Traderxyz/src/apis/nftswap.tsx

```diff
@@ -0,0 +1,12 @@
+import { NftSwap } from '@traderxyz/nft-swap-sdk'
+import { ExternalProvider, Web3Provider } from '@ethersproject/providers'
+import { getNetworkName, useWeb3Provider } from '@masknet/web3-shared-evm'
+
+export function getTraderApi(chainId: number) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const p = useWeb3Provider()
+    const network_name = getNetworkName(chainId)
+    const provider = new Web3Provider(p as unknown as ExternalProvider, chainId)
```

Comment:
Please do not create a new instance inside a hook directly. You should use `useRef`.

Reply:
can you provide some example in the repo?

Reply:
@guanbinrui

Reply:
Fixed, i have used this technique 
<img width="757" alt="image" src="https://user-images.githubusercontent.com/3579069/166129619-b0fe468c-a06d-4ff0-92e1-c4ccf34a3de7.png">